src/commands/git-metadata/gitdb.ts

```diff
@@ -116,13 +118,25 @@ const getLatestLocalCommits = async (git: simpleGit.SimpleGit) => {
 }
 
 const unshallowRepositoryWhenNeeded = async (log: Logger, git: simpleGit.SimpleGit) => {
-  const isShallow = (await git.revparse('--is-shallow-repository')) === 'true'
-  if (isShallow) {
+  const shallowCmdPromise = git.revparse('--is-shallow-repository')
+  const versionCmdPromise: simpleGit.Response<simpleGit.VersionResult> = git.version()
+  const isShallow = (await shallowCmdPromise) === 'true'
+  const gitversion = String(await versionCmdPromise)
+  if (isShallow && gte(gitversion, '2.27.0')) {
     log.info('[unshallow] Git repository is a shallow clone, unshallowing it...')
-    log.info('[unshallow] Setting remote.origin.partialclonefilter to "blob:none" to avoid fetching file content')
-    await git.addConfig('remote.origin.partialclonefilter', 'blob:none')
-    log.info(`[unshallow] Running git fetch --shallow-since="${MAX_HISTORY.oldestCommits}" --update-shallow --refetch`)
-    await git.fetch([`--shallow-since="${MAX_HISTORY.oldestCommits}"`, '--update-shallow', '--refetch'])
+    const headCmdPromise = git.revparse('HEAD')
+    const remoteNameCmdPromise = getDefaultRemoteName(git)
+    log.info(
+      `[unshallow] Running git fetch --shallow-since="${MAX_HISTORY.oldestCommits}" --update-shallow --filter="blob:none" --recurse-submodules=no`
+    )
+    await git.fetch([
+      `--shallow-since="${MAX_HISTORY.oldestCommits}"`,
+      '--update-shallow',
+      '--filter="blob:none"',
```

Comment:
this has bitten me in the past 😅 
```suggestion
      '--filter=blob:none',
```